enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,853 @@
+---
+title: aws-custom-edge-machineset-local-zones
+authors:
+  - "@mtulio"
+reviewers:
+  - "@rvanderp3"
+  - "@faermanj"
+  - "@patrickdillon"
+  - "@elmiko"
+  - "@JoelSpeed"
+  - "@sdodson"
+approvers:
+  - "@patrickdillon"
+  - "@JoelSpeed"
+  - "@sdodson"
+api-approvers:
+  - None
+creation-date: 2022-09-05
+last-updated: 2022-09-05
+tracking-link:
+  - https://issues.redhat.com/browse/RFE-2782
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Allow Customer-Provisioned edge machine pool for AWS Local Zones
+
+## Summary
+
+Edge adoption is increasing and providers are launching many offerings that fill
+that area. AWS launched the [Local Zones](https://aws.amazon.com/about-aws/global-infrastructure/localzones/)
+and the [Wavelength](https://aws.amazon.com/wavelength/features/) zones to extend the
+VPC network to the metropolitan area from big centers, and for Radio Access Network,
+respectively, to allow application developers to deliver low-latency applications
+at scale to the end users.
+
+As the resources in that locations were designed for specific workloads, and
+also have limited and more expensive cloud services, cluster operators should be
+able to create worker nodes on the edge networks on day zero operations.
+
+This enhancement proposes a new machine pool named `edge`, which will create
+Machine Sets for each availability zone on the AWS Local Zones to provision compute
+resources with `NoSchedule` taint.
+
+The EP is divided into three phases:
+
+- Phase 0: Research. Understand the Local Zones using the OKD/OCP, create a use case, and
+  provides the documentation of the installation of a cluster in an existing VPC.
+- Phase 1: Install a cluster in user provided network introducing to the installer a new
+  machine pool named `edge`. The MachineSet manifests will be created for each subnet
+  provided on `platform.aws.subnets` for Local Zones. The Machine will be protected with
+  taints to avoid running the normal platform workloads.
+- Phase 2: The installer should automate the network resource creation when the user
+  provides zone names with the type `local-zone` on the `platform.aws.zones` or
```

Comment:
> `platform.aws.zones`

nit: The `zones` field is only in the machine pools. There is a `defaultMachinePlatform` machine pool in `platform.aws` so there is `platform.aws.defaultMachinePlatform.zones`, but that would imply the zone would be used for both control plane and compute.

My understanding is we want to limit these edge zones to compute, so I think this should be:
```suggestion
  provides zone names with the type `local-zone` on the 
```

Reply:
@patrickdillon thanks for getting this. That's correct, it's not a goal to use `platform.aws.defaultMachinePlatform.zones` for `edge` compute pools.